Dockerfile

```diff
@@ -178,6 +182,23 @@ RUN --mount=type=cache,target=/.cache prototool format --overwrite --protoc-bin-
 FROM --platform=${BUILDPLATFORM} scratch AS fmt-protobuf
 COPY --from=proto-format-build /src/api/ /api/
 
+# run docgen for machinery config
+FROM build-go AS go-generate
+COPY ./pkg ./pkg
+COPY ./hack/boilerplate.txt ./hack/boilerplate.txt
+RUN --mount=type=cache,target=/.cache go generate ./pkg/...
+RUN goimports -w -local github.com/talos-systems/talos ./pkg/
+RUN gofumpt -w ./pkg/
+WORKDIR /src/pkg/machinery
+RUN --mount=type=cache,target=/.cache go generate ./...
+RUN gotagsrewrite .
+RUN goimports -w -local github.com/talos-systems/talos ./
+RUN gofumpt -w ./
+
+FROM go-generate AS gen-proto-go
+WORKDIR /src/
+RUN structprotogen github.com/talos-systems/talos/pkg/machinery/... /api/resource/definitions/
```

Comment:
should we add `--mount=type=cache,target=/.cache` to re-use build cache?

I wonder if we need same for `gotagsrewrite` ?

Reply:
gotagsrewrite doesn't use type information, only plain AST so it doesn't need build cache.

As for this - I dunno, it does use type checking in but I'm unsure if it uses go build cache. Will check.

Reply:
I checked - it does use `~/.cache` but I didn't notice any performance hit even with  empty `~/.cache/go-build` directory.